src/pykeen/models/base.py

```diff
@@ -138,10 +131,31 @@ def __init_subclass__(cls, **kwargs):
         if not inspect.isabstract(cls):
             parse_docdata(cls)
 
+    @property
+    def device(self) -> torch.device:
+        """Return the model's device."""
+        devices = self.get_devices()
+        if len(devices) == 0:
+            raise ValueError("Could not infer device, since there are neither parameters nor buffers.")
+        elif len(devices) > 1:
+            # prepare debug information
+            _info = defaultdict(list)
+            for name, tensor in itertools.chain(self.named_parameters(), self.named_buffers()):
+                _info[tensor.data.device].append(name)
+            info = {device: sorted(tensor_names) for device, tensor_names in _info.items()}
+            raise ValueError(f"Ambiguous device! Found: {devices}\n\n{info}")
+        else:
+            return next(iter(devices))
+
+    def get_devices(self) -> Collection[torch.device]:
+        """Return the device(s) from each components of the model."""
+        return {tensor.data.device for tensor in itertools.chain(self.parameters(), self.buffers())}
+
     def reset_parameters_(self):  # noqa: D401
         """Reset all parameters of the model and enforce model constraints."""
         self._reset_parameters_()
-        self.to_device_()
+        #  self.to(self.device)  # FIXME is this needed
```

Comment:
@mberr please check these fixmes

Reply:
I checked one of the. The empty-cache one is similar to the comment below: I think they shouldn't be necessary at all, but it seems like they have been in the past, so I did not modify this.